components/dialogs/ResourceUploadDialog.tsx

```
@@ -105,7 +106,21 @@ export function ResourceUploadDialog({
-        const response = await axios.get("/api/modules");
+
+        // Get session ID for anonymous users
+        const sessionId = getOrCreateSessionIdClient();
+
+        // Add sessionId to the request URL
+        let url = "/api/modules";
+        if (sessionId) {
+          url += `?sessionId=${sessionId}`;
+          console.log(
+            "Using sessionId for module fetch:",
+            sessionId.substring(0, 8) + "..."
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)